flopy/discretization/modeltime.py

```diff
@@ -71,21 +226,373 @@ def totim(self):
                 else:
                     dt = delt[-1] * tsmult
                 delt.append(dt)
+                per_stp.append((per, stp))
 
         totim = np.add.accumulate(delt)
-        return totim
+        self._totim_dict = {ps: totim[i] for i, ps in enumerate(per_stp)}
 
-    @property
-    def tslen(self):
-        n = 0
-        tslen = []
-        totim = self.totim
-        for ix, stp in enumerate(self.nstp):
-            for i in range(stp):
-                if not tslen:
-                    tslen = [totim[n]]
-                else:
-                    tslen.append(totim[n] - totim[n - 1])
-                n += 1
+    @staticmethod
+    def timeunits_from_user_input(units):
```

Comment:
If units moved out on their own this could be e.g. `TimeUnit.parse()`, but I dunno if worth it

Reply:
I'm not sure it's worth creating another class just for time unit parsing and storage

Reply:
maybe in flopy4 we can consider something like [pint](https://pint.readthedocs.io/en/stable/), although I'm not sure if that's worth it either.